src/main/kotlin/com/github/platan/tests_execution_chart/reporters/mermaid/TestExecutionMermaidDiagramFormatter.kt

```diff
@@ -11,10 +11,16 @@ class TestExecutionMermaidDiagramFormatter {
     fun format(report: TestExecutionScheduleReport): String {
         val diagramBuilder = MermaidGanttDiagram.MermaidGanttDiagramBuilder()
         report.results.forEach {
-            val testNameWithDuration = "${it.testName} - ${it.endTime.minus(it.startTime)} ms"
+            val testNameWithDuration = "${escape(it.testName)} - ${it.endTime.minus(it.startTime)} ms"
             diagramBuilder.add(it.className!!, testNameWithDuration, types[it.resultType], it.startTime, it.endTime)
         }
         val diagram = diagramBuilder.build("YYYY-MM-DD\\THH\\:mm\\:ss\\.SSSZ", "%H:%M:%S.%L")
         return MermaidGanttDiagramFormatter().format(diagram, "yyyy-MM-dd'T'HH:mm:ss.SSSZ")
     }
+
+    private fun escape(str: String): String {
```

Comment:
I think escaping should take place in `MermaidGanttDiagramFormatter`, because it's module responsible for creating output in mermaid format.

Reply:
Moved